image-rauc.c

```diff
@@ -52,6 +55,21 @@ static int rauc_generate(struct image *image)
 	if (ret)
 		goto out;
 
+	for (i = 0; i < cfg_size(image->imagesec, "intermediate"); i++) {
+		const char *uri;
+		int add = 1;
+
+		uri = cfg_getnstr(image->imagesec, "intermediate", i);
+		list_for_each_entry(part, &image->partitions, list) {
+			if (part->partition_type == RAUC_INTERMEDIATE
+					&& !strcmp(uri, part->image)) {
+				add = 0;
+				break;
+			}
+		}
+		if (add)
```

Comment:
Why not just
```suggestion
		if (!strncmp(pkcs11_prefix, url, strlen(pkcs11_prefix)))
```

Reply:
Because `pkcs11_prefix` is defined in `rauc_parse()`, so it is not available here.

I can move it to module space.

Reply:
Yes that's better.